### PR TITLE
Feature/signin with web

### PR DESCRIPTION
**Pull request by bajajcodes**
- Signin with web lets user login with otp code.
- On Screen `AuthScreen`, a button named `Sign In With Web` is added to login with web. 
   This button will open a component bottom sheet named `OtpModal` to enter otp code.
- `Sign In With Web` button is an resuable component for`AuthScreen`.
- `OtpModal` uses TextInput component for input and Text component for displaying otpcode as otpBoxes
    (each box displays a single code and toggles between either inactive and active based on user input).
-  `OtpModal` makes input focused on `onPress` event  on `OtpBoxes` components and 
     closes modal on clicking **back** button and overlay JSX element.
-   Utility function `isValidTextInput` is used for validating otpcode and allows otpcode as number or emptystring.
 

## Screen Recording
[feature_signin_with_web.mp4.webm](https://user-images.githubusercontent.com/29247011/192119804-ace28a6e-93ab-4129-9c3d-7f81b9a07ea9.webm)

### CODE REVIEW

**Comment by pranikz** [comment]
Looks good to me

**Review by pranikz — DISMISSED** [review]
Looks good to me

File: src/screens/AuthScreen/OtpModal/styles.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+export default StyleSheet.create({
+  container: {
+    position: 'relative',
+    width: '100%',
+    height: '100%',
+  },
+  backdrop: {
+    width: '100%',
+    height: '100%',
+    backgroundColor: 'rgba(0,0,0,0.5)',
```
Comment (Maheima) [inline src/screens/AuthScreen/OtpModal/styles.ts]: Can you please change this to hex code?

**Review by Maheima — DISMISSED** [review]

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 2]
```diff
@@ -15,15 +15,27 @@ import Images from '../../constants/images/Image';
...
 import { AuthViewStyle } from './styles';
+import { AuthScreenButton } from './Button';
+import { OtpModal } from './OtpModal';
 import { getUserData } from './Util';
 
 const AuthScreen = () => {
   const { setLoggedInUserData } = useContext(AuthContext);
   const [githubView, setGithubView] = useState(false);
+  const [otpCode, setOtpCode] = useState<string>('');
+  const [otpModalVisible, setotpModalVisible] = useState<boolean>(false);
```
Comment (poonammohata) [inline src/screens/AuthScreen/AuthScreen.tsx]: As we are using camelCase here in variable names, so shouldn't 'o' of 'otp' in the 'setotpModalVisible' variable be capital like it's in 'setOtpCode'?

File: src/screens/AuthScreen/OtpModal/styles.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: '#04148455',
+  },
+  submitButtonActive: {
+    backgroundColor: '#041484',
+  },
+  submitText: {
+    fontSize: 20,
+    color: 'white',
```
Comment (poonammohata) [inline src/screens/AuthScreen/OtpModal/styles.ts]: In most of the places hex codes are used for color, so can you please change this also to hex code?

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 4]
```diff
@@ -15,15 +15,27 @@ import Images from '../../constants/images/Image';
 import { storeData } from '../../utils/dataStore';
 import Strings from '../../i18n/en';
 import { AuthViewStyle } from './styles';
+import { AuthScreenButton } from './Button';
+import { OtpModal } from './OtpModal';
 import { getUserData } from './Util';
 
 const AuthScreen = () => {
   const { setLoggedInUserData } = useContext(AuthContext);
   const [githubView, setGithubView] = useState(false);
```
Comment (vinayak-trivedi) [inline src/screens/AuthScreen/AuthScreen.tsx]:
```suggestion
  const [githubView, setGithubView] = useState<boolean>(false);
```

File: src/screens/AuthScreen/Button/index.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Text, View, TouchableOpacity, Image } from 'react-native';
+import ButtonStyle from './styles';
+
+interface AuthScreenButtonProps {
+  text: string;
+  uri?: string;
+  onPress?: () => void;
+}
```
Comment (vinayak-trivedi) [inline src/screens/AuthScreen/Button/index.tsx]: can we use type here?

File: src/screens/AuthScreen/OtpModal/components.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import { View, Text } from 'react-native';
+import OtpModalStyle from './styles';
+
+interface OtpBoxProps {
+  index: number;
+  code: string;
+  maxLength: number;
+  textInputFocus: boolean;
+}
+
```
Comment (vinayak-trivedi) [inline src/screens/AuthScreen/OtpModal/components.tsx]: same as above can we use `type` here?

File: src/screens/AuthScreen/OtpModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  Keyboard,
+} from 'react-native';
+import { OtpBox } from './components';
+import OtpModalStyle from './styles';
+import { isValidTextInput } from '../Util';
+import Strings from '../../../i18n/en';
+
+interface OtpModalProps {
+  code: string;
+  maxLength: 4 | 6;
```
Comment (vinayak-trivedi) [inline src/screens/AuthScreen/OtpModal/index.tsx:17]: question: can we do it like this?

**Review by vinayak-trivedi — COMMENTED** [review]
Good work @shmbajaj. I've added some comments please have a look

File: src/screens/AuthScreen/OtpModal/index.tsx (hunk 7 above)
Comment (bajajcodes) [reply src/screens/AuthScreen/OtpModal/index.tsx:17]: I didn't understood.

**Review by shreya-mishra — APPROVED** [review]

**Review by pranikz — APPROVED** [review]
LGTM

**Pull request merged**